messytables/jsontableschema.py

```diff
@@ -0,0 +1,115 @@
+import json
+
+unicode = type(u'')
+
+class FormatError(Exception): pass
+class DuplicateFieldId(Exception): pass
+
+class JSONTableSchema(object):
+    """
+    A library to handle JSON Table Schema files.
+    See http://www.dataprotocols.org/en/latest/json-table-schema.html
+    Original source: https://pypi.python.org/pypi/json-table-schema
+    This remains licenced under the Apache Licence v2.0
+    Original by Martin Keegan - martin.keegan@okfn.org
+    Changed by David McKee - dave.mckee@gmail.com
+    Modified to make Python 3 compatible
```

Comment:
re: ilines - it's really not clear what it's actually achieving. Concerned it might be some edge case but no idea what it actually is. It appears copy-pasted from https://code.activestate.com/recipes/286165-ilines-universal-newlines-from-any-data-source/